mne_bids_pipeline/tests/test_run.py

```diff
@@ -278,3 +282,131 @@ def test_missing_sessions(
+                # For all other files, a simple copy suffices; rewriting
+                # `raw.info["subject_info"]["his_id"]` is not necessary because MNE-BIDS
+                # overwrites it with the value in `participants.tsv` anyway.
+                else:
+                    shutil.copyfile(
+                        src=walk_root / _file, dst=dst_dir / offset / bp.basename
+                    )
+    # emptyroom
+    src_dir = config_obj.bids_root / "sub-emptyroom"
+    dst_dir = new_bids_path.root / "sub-emptyroom"
+    shutil.copytree(src=src_dir, dst=dst_dir)
+    # root-level files (dataset description, etc)
+    src_dir = config_obj.bids_root
+    dst_dir = new_bids_path.root
+    files = [f for f in src_dir.iterdir() if f.is_file()]
+    for _file in files:
+        # in theory we should rewrite `participants.tsv` to remove the `sub-02` line,
+        # but in practice it will just get ignored so we won't bother.
+        shutil.copyfile(src=_file, dst=dst_dir / _file.name)
+    # derivatives (freesurfer files)
+    src_dir = config_obj.bids_root / "derivatives" / "freesurfer" / "subjects"
+    dst_dir = new_bids_path.root / "derivatives" / "freesurfer" / "subjects"
+    dst_dir.mkdir(parents=True)
+    freesurfer_subject_mapping = {"sub-01": "sub-01_ses-a", "sub-02": "sub-01_ses-b"}
+    for walk_root, dirs, files in src_dir.walk():
+        # change "root" so that in later steps of the walk when we're inside a subject's
+        # dir, the "offset" (folders between dst_dir and filename) will be correct
+        new_root = walk_root
+        if "sub-01" in walk_root.parts or "sub-02" in walk_root.parts:
+            new_root = Path(
+                *[freesurfer_subject_mapping.get(p, p) for p in new_root.parts]
+            )
+        offset = new_root.relative_to(src_dir)
+        # the actual subject dirs need their names changed
+        for _dir in dirs:
+            _dir = freesurfer_subject_mapping.get(_dir, _dir)
+            (dst_dir / offset / _dir).mkdir()
+        # for filenames that contain the subject identifier (BEM files, morph maps),
+        # we need to change the filename too, not just parent folder name
+        for _file in files:
+            dst_file = _file
+            for subj in freesurfer_subject_mapping:
+                if subj in dst_file:
+                    dst_file = dst_file.replace(subj, freesurfer_subject_mapping[subj])
+                    break
+            shutil.copyfile(src=walk_root / _file, dst=dst_dir / offset / dst_file)
+    # print_dir_tree(new_bids_path.root)  # for debugging
+    # update config so that `subjects_dir` and `deriv_root` also point to the tempdir
+    extra_config = f"""
+from pathlib import Path
+subjects_dir = "{new_bids_path.root / "derivatives" / "freesurfer" / "subjects"}"
+deriv_root = Path("{new_bids_path.root}") / "derivatives" / "mne-bids-pipeline" / "MNE-funloc-data"
+"""  # noqa E501
```

Comment:
It was not working to specify `deriv_root` as a `str`, hence the (IMO awkward) need to cast it to `Path` in this stringified `extra_config` (and also hence the inclusion of `Path` in `valid_extra_names` in `_config_import.py`).  If you're bothered by this, I can pass in `deriv_root` as a `str` and add in an explicit `Path()` cast wherever it's needed (I forget now exactly where the error occurred, but not hard to find out again)